client/login/index.js

```diff
@@ -17,7 +17,7 @@ export default router => {
 
 	if ( config.isEnabled( 'login/wp-login' ) ) {
 		router(
-			'/log-in/:twoFactorAuthType(authenticator|backup|sms|push)?/:socialConnect(social-connect)?/:lang?',
+			'/log-in/:twoFactorAuthType(authenticator|backup|sms|push)?/:socialConnect(social-connect)?/:privateSite(private-site)?/:lang?',
```

Comment:
If the goal is to use `http://calypso.localhost:3000/log-in/private-site` then we should not append this new [route parameter](https://expressjs.com/en/guide/routing.html#route-parameters) at the end of this [route path](https://expressjs.com/en/guide/routing.html#route-paths), as otherwise it makes urls such as http://calypso.localhost:3000/log-in/sms/private-site valid.

I think it's time to revisit this [route path](https://expressjs.com/en/guide/routing.html#route-paths) to avoid that sort of problems, and better separate concerns. This is something that can probably be achieved with:

```javascript
if ( config.isEnabled( 'login/magic-login' ) ) {
  router(
    [
      '/log-in/link/use/:lang?',
      '/log-in/link/:lang?'
    ],
    setUpLocale,
    redirectLoggedIn,
    magicLogin,
    makeLayout
  );

}

if ( config.isEnabled( 'login/wp-login' ) ) {
  router(
    [
      '/log-in/:twoFactorAuthType(authenticator|backup|sms|push)?/:lang?',
      '/log-in/:flow(social-connect|private-site)/:lang?',
    ],
    setUpLocale,
    login,
    makeLayout
  );
}
```